pkg/operator/staticpod/startupmonitor/startup_monitor.go

```diff
@@ -0,0 +1,370 @@
+package startupmonitor
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+// HealthChecker a contract between the startup monitor and operators
+// allows for initialization and an operand's health condition assessment
+type HealthChecker interface {
+	// Start is used to initialize and start the underlying health monitor
+	Start(context.Context) error
+
+	// IsTargetHealthy defines a function that abstracts away assessing operand's health condition.
+	// the provided functions should be async and cheap in a sense that it shouldn't assess the target
```

Comment:
why should it be cheap and async?

Reply:
because we hold a lock

Reply:
but then health must be indepedent of locking. Is it?